examples/hotrod/pkg/log/spanlogger.go

```diff
@@ -57,99 +58,164 @@ func (sl spanLogger) With(fields ...zapcore.Field) Logger {
 }
 
-func (fa *fieldAdapter) AddUint16(key string, value uint16) {
-	*fa = append(*fa, log.Uint64(key, uint64(value)))
+func (e *bridgeFieldEncoder) AddTime(key string, value time.Time) {
+	e.pairs = append(e.pairs, otTagToOTelAttr(key, value))
 }
 
-func (fa *fieldAdapter) AddUint8(key string, value uint8) {
-	*fa = append(*fa, log.Uint64(key, uint64(value)))
+func (e *bridgeFieldEncoder) AddUint(key string, value uint) {
+	e.pairs = append(e.pairs, otTagToOTelAttr(key, value))
 }
 
-func (fa *fieldAdapter) AddUintptr(key string, value uintptr)                        {}
-func (fa *fieldAdapter) AddArray(key string, marshaler zapcore.ArrayMarshaler) error { return nil }
-func (fa *fieldAdapter) AddComplex128(key string, value complex128)                  {}
-func (fa *fieldAdapter) AddComplex64(key string, value complex64)                    {}
-func (fa *fieldAdapter) AddObject(key string, value zapcore.ObjectMarshaler) error   { return nil }
-func (fa *fieldAdapter) AddReflected(key string, value interface{}) error            { return nil }
-func (fa *fieldAdapter) OpenNamespace(key string)                                    {}
+func (e *bridgeFieldEncoder) AddUint64(key string, value uint64) {
+	e.pairs = append(e.pairs, otTagToOTelAttr(key, value))
+}
 
-func (fa *fieldAdapter) AddDuration(key string, value time.Duration) {
-	// TODO inefficient
-	*fa = append(*fa, log.String(key, value.String()))
+func (e *bridgeFieldEncoder) AddUint32(key string, value uint32) {
+	e.pairs = append(e.pairs, otTagToOTelAttr(key, value))
 }
 
-func (fa *fieldAdapter) AddTime(key string, value time.Time) {
-	// TODO inefficient
-	*fa = append(*fa, log.String(key, value.String()))
+func (e *bridgeFieldEncoder) AddUint16(key string, value uint16) {
+	e.pairs = append(e.pairs, otTagToOTelAttr(key, value))
 }
 
-func (fa *fieldAdapter) AddBinary(key string, value []byte) {
-	*fa = append(*fa, log.Object(key, value))
+func (e *bridgeFieldEncoder) AddUint8(key string, value uint8) {
+	e.pairs = append(e.pairs, otTagToOTelAttr(key, value))
 }
 
-func (fa *fieldAdapter) AddByteString(key string, value []byte) {
-	*fa = append(*fa, log.Object(key, value))
+func (e *bridgeFieldEncoder) AddUintptr(key string, value uintptr) {
+	e.pairs = append(e.pairs, otTagToOTelAttr(key, value))
 }
 
-func (fa *fieldAdapter) AddString(key, value string) {
-	if key != "" && value != "" {
-		*fa = append(*fa, log.String(key, value))
+func (e *bridgeFieldEncoder) AddReflected(key string, value interface{}) error { return nil }
+func (e *bridgeFieldEncoder) OpenNamespace(key string)                         {}
+
+// otTagToOTelAttr converts given key-value into attribute.KeyValue.
+// Note that some conversions are not obvious:
+// - int -> int64
+// - uint -> string
+// - int32 -> int64
+// - uint32 -> int64
+// - uint64 -> string
+// - float32 -> float64
+//
+// See: https://github.com/open-telemetry/opentelemetry-go/blob/main/bridge/opentracing/bridge.go#L538
```

Comment:
As the comment states, the mappings below are taken from the OTEL bridge to maintain consistency.